src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -131,6 +135,21 @@ public HashSet(IEnumerable<T> collection, IEqualityComparer<T> comparer)
             }
         }
 
+        public HashSet(int capacity, IEqualityComparer<T> comparer)
+            : this(comparer)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException("capacity");
+            }
+            Contract.EndContractBlock();
```

Comment:
It is something definitely worth considering, to this point I've not seen the value of ripping it all out yet. It may still be somewhat useful if we do get the C# vNext contracts feature in providing some hints about the contracts.